### PR TITLE
Refactor sidebar renders into methods (prep for mwc-list conversion)

### PR DESCRIPTION
**Pull request by donkawechico**
## Proposed change

The sidebar render code is large and can be a bit difficult to reason about. This PR extracts the various "sections" of sidebar into their own render methods for better readability.

This is the first step toward converting sidebar to using mwc-list instead of paper-listbox. Related PR: https://github.com/home-assistant/frontend/pull/7326

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/home-assistant/frontend/issues/7442
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-sidebar.ts [hunk 1]
```diff
@@ -202,198 +202,227 @@ class HaSidebar extends LitElement {
...
-    const hass = this.hass;
-
-    if (!hass) {
+    if (!this.hass) {
       return html``;
     }
 
+    return html`
+      ${this.renderHeader()} ${this.renderItems()} ${this.renderDivider()}
+      ${this.renderNotifications()} ${this.renderUserItem()}
```
Comment (bramkragten) [inline src/components/ha-sidebar.ts:211]:
```ts
      ${this.renderHeader()}
      ${this.renderPanels()}
      ${this.renderDivider()}
      ${this.renderNotificationItem()}
      ${this.renderUserItem()}
```

File: src/components/ha-sidebar.ts [hunk 2]
```diff
@@ -202,198 +202,227 @@ class HaSidebar extends LitElement {
...
+      <div disabled class="bottom-spacer"></div>
+      <div class="tooltip"></div>
+    `;
+  }
+
+  protected renderHeader() {
```
Comment (bramkragten) [inline src/components/ha-sidebar.ts:217]: Let's make these private.

**Comment by bramkragten** [comment]
You didn't change any actual logic here right?

File: src/components/ha-sidebar.ts (hunk 1 above)
Comment (donkawechico) [reply src/components/ha-sidebar.ts:211]: It auto-prettifies to this :( :( 

I can work around it by pulling the wrapping <div>s out of those methods and surrounding the call, but that doesn't feel right to me -- a `renderX()` should be fully responsible for rendering its content.

**Comment by donkawechico** [comment]
> You didn't change any actual logic here right?

95% true. I arguably "changed the logic" with how items in edit mode are rendered. That is, I converted this flattened, in-line pattern:

```
protected oneRenderToRuleThemAll() {
  return html`
    <div>Some stuff</div>
    ${this.isEditMode ? "Render items in edit mode" : "Render items in normal mode"}
    <div>Some other stuff</div>
  `
}
```

And refactored into state pattern:

```
protected renderItems() {
  return html`
    this.editMode ? this.renderEditModeItems() : this.renderNormalModeItems()
  `;
}

protected renderEditModeItems() {
  return html`
       <div>Some stuff</div>
      Render items in edit mode
      <div>Some other stuff</div>
  `;
}

protected renderNormalMode() {
  return html`
       <div>Some stuff</div>
      Render items in normal mode
      <div>Some other stuff</div>
  `;
}
```

File: src/components/ha-sidebar.ts (hunk 2 above)
Comment (donkawechico) [reply src/components/ha-sidebar.ts]: Done

File: src/components/ha-sidebar.ts (hunk 1 above)
Comment (donkawechico) [reply src/components/ha-sidebar.ts]: Disabled prettier in two places.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**